security/auto-host-endpoints.md

```diff
@@ -0,0 +1,263 @@
+```
+
+If successful, host endpoints are created for each of your cluster's nodes:
+
+```bash
+calicoctl get heps -owide
+```
+
+The output may look similar to this:
+
+```
+$ calicoctl get heps -owide
+NAME                                                    NODE                                           INTERFACE   IPS                              PROFILES
+ip-172-16-101-147.us-west-2.compute.internal-auto-hep   ip-172-16-101-147.us-west-2.compute.internal   *           172.16.101.147,192.168.228.128   projectcalico-default-allow
+ip-172-16-101-54.us-west-2.compute.internal-auto-hep    ip-172-16-101-54.us-west-2.compute.internal    *           172.16.101.54,192.168.107.128    projectcalico-default-allow
+ip-172-16-101-79.us-west-2.compute.internal-auto-hep    ip-172-16-101-79.us-west-2.compute.internal    *           172.16.101.79,192.168.91.64      projectcalico-default-allow
+ip-172-16-101-9.us-west-2.compute.internal-auto-hep     ip-172-16-101-9.us-west-2.compute.internal     *           172.16.101.9,192.168.71.192      projectcalico-default-allow
+ip-172-16-102-63.us-west-2.compute.internal-auto-hep    ip-172-16-102-63.us-west-2.compute.internal    *           172.16.102.63,192.168.108.192    projectcalico-default-allow
+```
+
+#### Apply network policy to automatic host endpoints
+
+Automatic host endpoints share the common label `projectcalico.org/created-by: calico-kube-controllers`.
+To write policy that targets all Kubernetes nodes, use that label:
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: all-nodes-policy
+spec:
+  selector: projectcalico.org/created-by == 'calico-kube-controllers'
+  <rest of the policy>
+```
+
+To select a specific set of host endpoints (and their corresponding Kubernetes nodes), use a policy selector that selects a label unique to that set of host endpoints.
+We can take advantage of the syncing of node labels to automatic host endpoints and label that set of nodes with a unique key/value pair.
+For example, if we want to add the label `environment=dev` to nodes named `node1` and `node2`:
+
+```bash
+kubectl label node node1 environment=dev
+kubectl label node node2 environment=dev
+```
+
+With the labels in place and automatic host endpoints enabled, hostendpoints for node1 and node2 will be updated with the `environment=dev` label.
+We can write policy to select that set of nodes with a combination of selectors:
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: all-nodes-policy
+spec:
+  selector: projectcalico.org/created-by == 'calico-kube-controllers' && environment == 'dev'
+  <rest of the policy>
+```
+
+#### Restrict Kubernetes node ingress
+
+This tutorial will lock down Kubernetes node ingress to only allow SSH and required ports for Kubernetes to function.
+
+> Note: Do not run this tutorial on a cluster that is important. This tutorial may disrupt traffic in your cluster.
+> Only run this tutorial on a sandbox cluster.
```

Comment:
Note: Run this tutorial only on a sandbox cluster; using it on a real cluster can disrupt traffic.